chips/nrf52832/src/interrupt_service.rs

```diff
@@ -1,21 +1,30 @@
-use crate::gpio;
-use nrf52::interrupt_service::InterruptService;
+use crate::deferred_call_tasks::DeferredCallTask;
+use nrf52::chip::Nrf52DefaultPeripherals;
 
-pub struct Nrf52832InterruptService {
-    nrf52: nrf52::interrupt_service::Nrf52InterruptService<'static>,
+/// This struct, when initialized, instantiates all peripheral drivers for the nrf52840.
+/// If a board wishes to use only a subset of these peripherals, this
+/// should not be used or imported, and a modified version should be
+/// constructed manually in main.rs.
+pub struct Nrf52832DefaultPeripherals<'a> {
+    pub nrf52_base: Nrf52DefaultPeripherals<'a>,
+    // put additional 52832 specific peripherals here
 }
-
-impl Nrf52832InterruptService {
-    pub unsafe fn new() -> Nrf52832InterruptService {
-        Nrf52832InterruptService {
-            nrf52: nrf52::interrupt_service::Nrf52InterruptService::new(&gpio::PORT),
+impl<'a> Nrf52832DefaultPeripherals<'a> {
+    pub fn new(ppi: &'a crate::ppi::Ppi) -> Self {
+        Self {
+            nrf52_base: unsafe { Nrf52DefaultPeripherals::new(&crate::gpio::PORT, ppi) },
```

Comment:
Does hard-coding `crate::gpio::PORT` here fits in the new model introduced in this pull request?

Reply:
It does not really fit the model. Unfortunately, keeping a single `Port` type despite differing number of pins between boards is not very ergonomic without `min_const_generics`, so I was planning to wait on that to stabilize rather than transition to something where we continue using slices but stores the pins separately from the `Port`.

Reply:
Makes sense. We could mention that in a comment in the code, so that later we think about it when `min_const_generics` gets more stable.

Reply:
I added a comment in both places indicating that this is waiting on `min_const_generics`